ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/update_available/AppUpdateCheckerImp.kt

```diff
@@ -0,0 +1,133 @@
+     */
+    override fun checkForUpdateAvailability(
+        context: Context,
+        stalenessDays: Int
+    ): Flow<UpdateInfo> = callbackFlow {
+        val appUpdateInfoTask = AppUpdateManagerFactory.create(context.applicationContext).appUpdateInfo
+
+        appUpdateInfoTask.addOnCompleteListener { infoTask ->
+            if (!infoTask.isSuccessful) {
+                emitFailure(this)
+                return@addOnCompleteListener
+            }
+
+            val appUpdateInfo = infoTask.result
+            if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE &&
+                appUpdateInfo.isUpdateTypeAllowed(AppUpdateType.IMMEDIATE)
+            ) {
+                // we force user to update immediately in case of high priority
+                // or in case of staleness days passed
+                if (isHighPriority(appUpdateInfo.updatePriority()) ||
+                    (appUpdateInfo.clientVersionStalenessDays() ?: -1) >= stalenessDays
+                ) {
+                    emitSuccess(this, infoTask.result, UpdateState.Prepared)
+                } else {
+                    emitSuccess(this, infoTask.result, UpdateState.Done)
+                }
+            }
+        }
+        awaitClose {
+            // No resources to release
+        }
+    }
+
+    private fun emitSuccess(producerScope: ProducerScope<UpdateInfo>, info: AppUpdateInfo, state: UpdateState) {
+        producerScope.trySend(
+            UpdateInfoFixture.new(
+                getPriority(info.updatePriority()),
+                isHighPriority(info.updatePriority()),
+                info,
+                state
+            )
+        )
+    }
+
+    private fun emitFailure(producerScope: ProducerScope<UpdateInfo>) {
+        producerScope.trySend(
+            UpdateInfoFixture.new(
+                AppUpdateChecker.Priority.LOW,
+                false,
+                null,
+                UpdateState.Failed
+            )
+        )
+    }
+
+    /**
+     * This function is used for triggering in-app update with IMMEDIATE app update type.
+     *
+     * The immediate update can result with these values:
+     * Activity.RESULT_OK: The user accepted and the update succeeded (which, in practice, your app
+     * never should never receive because it already updated).
+     * Activity.RESULT_CANCELED: The user denied or canceled the update.
+     * ActivityResult.RESULT_IN_APP_UPDATE_FAILED: The flow failed either during the user confirmation,
+     * the download, or the installation.
+     *
+     * @param activity
+     * @param appUpdateInfo object is necessary for starting the update process,
+     * for getting it see {@link #checkForUpdateAvailability()}
+     * @param onUpdateResult callback for sending resultCode to the caller
+     *
+     */
+    override fun startUpdate(
+        activity: ComponentActivity,
+        appUpdateInfo: AppUpdateInfo?
```

Comment:
It's nullable because the AppUpdateInfo field of the UpdateInfo object has to be nullable, as it's filled after we get it from the API. But yes, we should move the nullable check a bit. I've just made a change to this.